app/src/main/java/de/tu_darmstadt/seemoo/nfcgate/nfc/chip/detectors/STDetector.java

```diff
@@ -26,20 +26,20 @@ protected boolean onLine(String line, NfcChipGuess guess) {
         Pair<String, String> keyVal = splitConfigLine(line);
 
         if (keyVal != null) {
-            if ("NCI_HAL_MODULE".equals(keyVal.first)) {
-                String device = "/dev/" + keyVal.second.replace("nfc_nci.", "");
-                // existence of this device node confirms this is (or is not) the correct config
-                if (!fileExists(device) || "/dev/null".equals(device))
-                    return false;
-
-                guess.confidence = 0.9f;
-                if (guess.chipName == null)
-                    guess.chipName = "ST Device " + formatSTDeviceNode(keyVal.second);
-            }
-            else if ("STNFC_FW_BIN_NAME".equals(keyVal.first)
+            if ("STNFC_FW_BIN_NAME".equals(keyVal.first)
                     || "STNFC_FW_CONF_NAME".equals(keyVal.first)) {
                 guess.improveConfidence(0.2f);
-                guess.chipName = "NXP Device " + formatFirmwareName(keyVal.second);
+                guess.chipName = "ST Device " + formatFirmwareName(keyVal.second);
+            }
+            else if ("NCI_HAL_MODULE".equals(keyVal.first)) {
+                String module = keyVal.second.replace("nfc_nci.", "");
+                String device = "/dev/" + module.split("\\.")[0];
+                // existence of this device node confirms this is (or is not) the correct config
+                if (fileExists(device) && !"/dev/null".equals(device)) {
+                    guess.confidence = 0.9f;
+                    if (guess.chipName == null)
+                        guess.chipName = "ST Device " + formatSTDeviceNode(keyVal.second);
+                }
```

Comment:
Please revert this to the "early-return in error case" style, as used everywhere in the repo.
```suggestion
                if (!fileExists(device) || "/dev/null".equals(device))
                    return false;

                guess.confidence = 0.9f;
                if (guess.chipName == null)
                    guess.chipName = "ST Device " + formatSTDeviceNode(keyVal.second);
```

Reply:
Done.